src/fairseq2/models/wav2vec2/builder.py

```diff
@@ -145,6 +158,9 @@ class Wav2Vec2EncoderConfig:
     conv_norm_type: Literal["batch_norm", "layer_norm"]
     """The type of norm layer in the Conformer convolution module."""
 
+    shaw_rel_position_sdpa_config: Optional[ShawRelativePositionSDPAConfig]
```

Comment:
Nice, it would be a BC breaking change, but I think it would be nicer if we could wrap parameters for "conv" positional encoder in the same way in a dataclass. Not relevant for this PR  though. Just food for thought :)

Reply:
Could you elaborate `wrap parameters for "conv" positional encoder in the same way in a dataclass`? I don't understand at all.

Reply:
It is not related to this PR at all. In `Wav2Vec2EncoderConfig` we have several attributes related to the default convolutional position encoder. We can consolidate them under a single configuration dataclass like you did for Shaw encoder.

Reply:
Yes we should do that, even I was thinking of that.